roofit/roostats/inc/RooStats/ProposalFunction.h

```diff
@@ -65,19 +65,15 @@ the proposal density to maintain detailed balance.
       /// Return true if all objects are RooRealVars, false otherwise
       virtual bool CheckParameters(RooArgSet& params)
       {
-         TIterator* it = params.createIterator();
-         TObject* obj;
-         while ((obj = it->Next()) != NULL) {
+         for (auto *obj : static_range_cast<TObject *>(params)){
```

Comment:
```suggestion
         for (auto *obj : params){
```
Unnecessary cast I think (the objects are `RooAbsArg` which already inherits from `TObject`.